sphinx/screens/chats/chat-common/chat-common/src/main/java/chat/sphinx/chat_common/ui/ChatFragment.kt

```diff
@@ -711,6 +751,58 @@ abstract class ChatFragment<
             }
         }
 
+        onStopSupervisor.scope.launch(viewModel.mainImmediate) {
+            viewModel.getAttachmentFullscreenViewStateFlow().collect { viewState ->
+                attachmentFullscreenBinding.apply {
+                    @Exhaustive
+                    when (viewState) {
+                        is AttachmentFullscreenViewState.Idle -> {
+                            root.gone
+                            imageViewAttachmentFullscreen.setImageDrawable(null)
+                        }
+                        is AttachmentFullscreenViewState.Fullscreen -> {
+                            imageViewAttachmentFullscreen.scaleX = 1.0f
+                            imageViewAttachmentFullscreen.scaleY = 1.0f
+                            imageViewAttachmentFullscreen.translationX = 0f
+                            imageViewAttachmentFullscreen.translationY = 0f
+
+                            if (viewState.media?.localFile != null) {
+                                imageLoader.load(
+                                    imageViewAttachmentFullscreen,
+                                    viewState.media.localFile!!
+                                )
+                            } else {
+                                val builder = ImageLoaderOptions.Builder()
+
+                                viewState.media?.host?.let { host ->
+                                    viewModel.memeServerTokenHandler.retrieveAuthenticationToken(host)?.let { token ->
+                                            builder.addHeader(token.headerKey, token.headerValue)
+
+                                            viewState.media.mediaKeyDecrypted?.value?.let { key ->
+                                                val header = CryptoHeader.Decrypt.Builder()
+                                                    .setScheme(CryptoScheme.Decrypt.JNCryptor)
+                                                    .setPassword(key)
+                                                    .build()
+
+                                                builder.addHeader(header.key, header.value)
+                                            }
+                                        }
+                                }
+                                imageLoader.load(
+                                    imageViewAttachmentFullscreen,
+                                    viewState.url,
+                                    builder.build()
+                                )
```

Comment:
this should be launched by a new coroutine not utilizing onStopSupervisor who's job + disposable are set to a variable where by if the state changes before retrieval and loading is complete they are canceled.

there would also need to be a logging of current state on the Fragment that get cleared in onDestroyView to inhibit re-execution upon re-collection if the state has not changed when the user puts the application into the background and brings it back to the foreground.